presto-rcfile/src/test/java/com/facebook/presto/rcfile/TestRcFileReaderManual.java

```diff
@@ -0,0 +1,333 @@
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.lang.Math.toIntExact;
+import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertEquals;
+
+public class TestRcFileReaderManual
+{
+    private static final Slice COLUMN_COUNT_METADATA_KEY = utf8Slice("hive.io.rcfile.column.number");
+    private static final Slice RCFILE_MAGIC = utf8Slice("RCF");
+    private static final int CURRENT_VERSION = 1;
+    private static final long syncFirst = 0x1234_5678_9012_3456L;
+    private static final long syncSecond = 0x7890_1234_5678_9012L;
+
+    @Test
+    public void testNoStartSync()
+            throws Exception
+    {
+        SliceOutput output = new DynamicSliceOutput(10 * 1024);
+
+        List<Segment> segments = ImmutableList.of(
+                writeSegment(output, ImmutableList.of(ImmutableList.of(0, 2, 3, 4), ImmutableList.of(10, 12, 13))),
+                writeSegment(output, ImmutableList.of(ImmutableList.of(20, 22), ImmutableList.of(30, 33), ImmutableList.of(40, 44))),
+                writeSegment(output, ImmutableList.of(ImmutableList.of(100, 101, 102))));
+
+        assertFileSegments(output.slice(), segments);
+    }
+
+    @Test
+    public void testStartSync()
+            throws Exception
+    {
+        SliceOutput output = new DynamicSliceOutput(10 * 1024);
+
+        List<Segment> segments = ImmutableList.of(
+                writeSegment(output, ImmutableList.of()),
+                writeSegment(output, ImmutableList.of(ImmutableList.of(0, 2, 3, 4), ImmutableList.of(10, 12, 13))),
+                writeSegment(output, ImmutableList.of(ImmutableList.of(20, 22), ImmutableList.of(30, 33), ImmutableList.of(40, 44))),
+                writeSegment(output, ImmutableList.of(ImmutableList.of(100, 101, 102))));
+
+        assertFileSegments(output.slice(), segments);
+    }
+
+    private static void assertFileSegments(Slice file, List<Segment> segments)
+            throws IOException
+    {
+        // read whole file
+        List<Integer> allValues = segments.stream()
+                .map(Segment::getValues)
+                .flatMap(List::stream)
+                .collect(toList());
+        assertEquals(allValues, readValues(file, 0, file.length()));
+
+        for (int segmentIndex = 0; segmentIndex < segments.size(); segmentIndex++) {
+            Segment segment = segments.get(segmentIndex);
+
+            // whole segment
+            assertEquals(segment.getValues(), readValues(file, segment.getOffset(), segment.getLength()));
+            // first byte of segment
+            assertEquals(segment.getValues(), readValues(file, segment.getOffset(), 1));
+            // straddle segment start
+            assertEquals(segment.getValues(), readValues(file, segment.getOffset() - 1, 2));
+
+            // regions entirely within the the segment
+            assertEquals(ImmutableList.of(), readValues(file, segment.getOffset() + 1, 1));
```

Comment:
Add `assertEquals(ImmutableList.of(), readValues(file, segment.getOffset() + 1, segment.getLength() - 1));`